### PR TITLE
Split RichUpdateReport into internal

### PR DESCRIPTION
**Pull request by eed3si9n**
The main change is d6cadf79171875798f3d9c92d52352f6d76526bf.
This is so we can use the injected methods on `UpdateReport` from the scripted tests. See https://github.com/sbt/sbt/pull/2226#issuecomment-144661630

/review @Duhemm, @dwijnand, @jsuereth

### CODE REVIEW

File: deltag.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+for t in `git tag`
+do
+  git tag -d $t
+done
+
```
Comment (dwijnand) [inline deltag.sh]: woops?

Comment (eed3si9n) [inline deltag.sh]: oops

**Comment by dwijnand** [comment]
LGTM pending Travis

**Comment by Duhemm** [comment]
Was this already included in 0.1.0-M4? Any reason why it has not been merged?

**Comment by eed3si9n** [comment]
0.1.0-M4 is c97563a0c0df4c0c35421ed33e78495ab6e50dc4.
I'll merge this now.

**Pull request merged**